Lib/http/cookiejar.py

```diff
@@ -1761,11 +1762,8 @@ def __init__(self, filename=None, delayload=False, policy=None):
 
         """
         CookieJar.__init__(self, policy)
-        if filename is not None:
-            try:
-                filename+""
-            except:
-                raise ValueError("filename must be string-like")
+        if filename is not None and not isinstance(filename, (str, pathlib.Path)):
```

Comment:
There could be other objects that can implement `__fspath__` so using `os.PathLike` seems more robust to me https://docs.python.org/3/library/os.html#os.PathLike

Reply:
updated